client/my-sites/marketing/connections/account-dialog.jsx

```diff
@@ -89,9 +89,11 @@ class AccountDialog extends Component {
 	}
 
 	areAccountsConflicting( account, otherAccount ) {
+		// Accounts are conflicting, if they have the same ID and keyringConnectionId, which
+		// means they are the same account. Otherwise it's a different site for the same account.
 		return (
```

Comment:
This should be behind the feature flag as well, so we don't change the behaviour for existing accounts.